cpp/Platform.Ranges/Range[T].h

```diff
@@ -2,73 +2,94 @@
     {
         template<typename TArgument>
         void MaximumArgumentIsGreaterOrEqualToMinimum(TArgument&& minimumArgument, TArgument&& maximumArgument, const std::string& maximumArgumentName);
     }
 
     template <typename ...> struct Range;
-    template <Internal::ranged T> struct Range<T>
+    template <std::three_way_comparable T> struct Range<T>
     {
         public: const T Minimum;
 
         public: const T Maximum;
 
-        public: explicit Range(T minimumAndMaximum) noexcept
-            : Minimum(std::move(minimumAndMaximum)), Maximum(std::move(minimumAndMaximum))
+        public: constexpr explicit Range(T minimumAndMaximum) noexcept
+            : Minimum(std::move(minimumAndMaximum)), Maximum(Minimum)
         {
         }
 
         public: constexpr Range(T minimum, T maximum)
             : Minimum(std::move(minimum)), Maximum(std::move(maximum))
         {
-            if (minimum > maximum) // for constexpr support
+            if (Minimum > Maximum) // for constexpr support
             {
-                Always::MaximumArgumentIsGreaterOrEqualToMinimum(minimum, maximum, "maximum");
+                Ensure::Always::MaximumArgumentIsGreaterOrEqualToMinimum(minimum, maximum, "maximum");
             }
         }
 
-        public: explicit operator std::string() const { return std::string("[").append(Converters::To<std::string>(Minimum)).append(", ").append(Converters::To<std::string>(Maximum)).append(1, ']'); }
+        public: explicit operator std::string() const requires Internal::formattable<T> { return std::string("[").append(Converters::To<std::string>(Minimum)).append(", ").append(Converters::To<std::string>(Maximum)).append(1, ']'); }
 
         public: friend std::ostream& operator<<(std::ostream& out, const Range<T>& obj) { return out << (std::string)obj; }
 
-        public: constexpr bool Contains(T value) noexcept { return Minimum <= value && Maximum >= value; }
-
-        public: constexpr bool Contains(const Range<T>& range) noexcept { return Contains(range.Minimum) && Contains(range.Maximum); }
+        public: [[nodiscard]] constexpr bool Contains(T value) const noexcept { return Minimum <= value && Maximum >= value; }
 
-        public: constexpr bool operator==(const Range<T> &other) const noexcept { return Minimum == other.Minimum && Maximum == other.Maximum; }
+        public: [[nodiscard]] constexpr bool Contains(const Range<T>& range) const noexcept { return Contains(range.Minimum) && Contains(range.Maximum); }
 
         public: constexpr explicit operator std::tuple<T, T>() const noexcept { return { Minimum, Maximum }; }
 
         public: constexpr explicit Range(std::tuple<T, T> tuple) noexcept : Range(std::get<0>(tuple), std::get<1>(tuple))  { }
 
-        // TODO: @Voider extension
+
+    public: // TODO: @Voider extensions
+        template<typename TOther> requires std::equality_comparable_with<T, TOther>
+        constexpr bool operator==(const Range<TOther>& other) const noexcept
```

Comment:
Please move this operator to the previous position (the single line style should be preserved).